src/Datadog.Trace/AppSec/Waf/NativeBindings/Native.cs

```diff
@@ -0,0 +1,404 @@
+        {
+            // first get anything "home folder" like, then combine with the profiler's location
+            // taking into account that this should be the same place
+
+            var paths = GetHomeFolders(runtimeId);
+            var profilerFolder = GetProfilerFolder(frameworkDescription);
+            paths.Add(profilerFolder);
+
+            return paths.Distinct().ToList();
+        }
+
+        private static string GetProfilerFolder(FrameworkDescription frameworkDescription)
+        {
+            var profilerEnvVar =
+                frameworkDescription.IsCoreClr() ? "CORECLR_PROFILER_PATH" : "COR_PROFILER_PATH";
+            var profilerEnvVarBitsExt =
+                profilerEnvVar + (Environment.Is64BitProcess ? "_64" : "_32");
+
+            var profilerPathsEnvVars = new List<string>()
+            {
+                profilerEnvVarBitsExt,
+                profilerEnvVar
+            };
+
+            // it is unlikely that the security library would be in a sub folder from
+            // where the profiler lives, so just use this paths directly
+            var profilerFolders =
+                profilerPathsEnvVars
+                    .Select(Environment.GetEnvironmentVariable)
+                    .Where(x => !string.IsNullOrWhiteSpace(x))
+                    .Select(Path.GetDirectoryName)
+                    .FirstOrDefault();
+
+            return profilerFolders;
+        }
+
+        private static List<string> GetHomeFolders(string runtimeId)
+        {
+            List<string> GetRuntimeSpecificVersions(string path)
+            {
+                return new List<string>()
+                {
+                    path,
+                    Path.Combine(path, runtimeId),
+                    Path.Combine(path, runtimeId, "native"),
+                };
+            }
+
+            // treat any path that could contain integrations.json as home folder
+            var integrationsPaths = Environment.GetEnvironmentVariable("DD_INTEGRATIONS")
+                    ?.Split(';')
+                    ?.Where(x => !string.IsNullOrWhiteSpace(x))
+                    ?.Select(Path.GetDirectoryName)
+                    ?.ToList()
+                        ?? new List<string>();
+
+            // the real trace home
+            var tracerHome = Environment.GetEnvironmentVariable("DD_DOTNET_TRACER_HOME");
+            if (!string.IsNullOrWhiteSpace(tracerHome))
+            {
+                integrationsPaths.Add(tracerHome);
+            }
+
+            // include the appdomain base as this will help framework samples running in IIS find the library
+            var currentDir =
+                string.IsNullOrEmpty(AppDomain.CurrentDomain.RelativeSearchPath) ?
+                    AppDomain.CurrentDomain.BaseDirectory : AppDomain.CurrentDomain.RelativeSearchPath;
+            integrationsPaths.Add(currentDir);
+
+            Log.Warning($"current dir is {currentDir}'");
```

Comment:
If you're still using this for debugging purposes, maybe this can be a Debug level log

Reply:
fixed in 56eff08dc8899e65b62d0b8786e3b281851090b4